runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java

```diff
@@ -288,37 +319,69 @@ else if (menuAction == MenuAction.EXAMINE_NPC && client.isKeyPressed(KeyCode.KC_
 	public void onMenuOptionClicked(MenuOptionClicked click)
 	{
 		if (click.getMenuAction() != MenuAction.RUNELITE ||
-			!(click.getMenuOption().equals(TAG) || click.getMenuOption().equals(UNTAG)))
+			(!(click.getMenuOption().equals(TAG) || click.getMenuOption().equals(UNTAG)) &&
+			(!(click.getMenuOption().equals(TAG_ALL) || click.getMenuOption().equals(UNTAG_ALL)))))
 		{
 			return;
 		}
 
-		final int id = click.getId();
-		final boolean removed = npcTags.remove(id);
-		final NPC[] cachedNPCs = client.getCachedNPCs();
-		final NPC npc = cachedNPCs[id];
-
-		if (npc == null || npc.getName() == null)
+		if (click.getMenuOption().equals(TAG) || click.getMenuOption().equals(UNTAG))
 		{
-			return;
-		}
+			final int id = click.getId();
+			final boolean removed = npcTags.remove(id);
+			final NPC[] cachedNPCs = client.getCachedNPCs();
+			final NPC npc = cachedNPCs[id];
 
-		if (removed)
-		{
-			highlightedNpcs.remove(npc);
-			memorizedNpcs.remove(npc.getIndex());
+			if (npc == null || npc.getName() == null)
+			{
+				return;
+			}
+
+			if (removed)
+			{
+				highlightedNpcs.remove(npc);
+				memorizedNpcs.remove(npc.getIndex());
+			}
+			else
+			{
+				if (!client.isInInstancedRegion())
+				{
+					memorizeNpc(npc);
+					npcTags.add(id);
+				}
+				highlightedNpcs.add(npc);
+			}
+
+			click.consume();
 		}
 		else
 		{
-			if (!client.isInInstancedRegion())
+			final String npcNames = config.getNpcToHighlight().toLowerCase();
+			List<String> highlightNames = new ArrayList<>();
+
+			if (!npcNames.trim().isEmpty())
 			{
-				memorizeNpc(npc);
-				npcTags.add(id);
+				highlightNames = new ArrayList<>(Arrays.asList(npcNames.split(",")));
 			}
-			highlightedNpcs.add(npc);
-		}
 
-		click.consume();
+			final int id = click.getId();
```

Comment:
These 3 lines + the conditionals below are duplicated in both branches, so move them to be above.